src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java

```diff
@@ -42,8 +42,16 @@ public void addSupplementaryDatasetEntity(SupplementaryDatasetDTO supplementaryD
       supplementaryDatasetRepository.save(supplementaryDatasetEntity);
       log.info("Successfully saved the supplementary dataset to the database");
     } catch (Exception e) {
-      throw new CTPException(
-          CTPException.Fault.SYSTEM_ERROR, "Something went wrong adding dataset", e);
+      if (collectionExercise == null) {
+        log.error(
+            "Failed to find collection exercise for supplementary dataset. "
+                + "SurveyId: {}, PeriodId: {}",
+            supplementaryDatasetDTO.getSurveyId(),
+            supplementaryDatasetDTO.getPeriodId());
+        throw new CTPException(
```

Comment:
Have we lost the original handling for the other (`SYSTEM_ERROR`) exceptions that may occur in this try block (there are also SDS database activity happening that might go wrong)?

It might be worth leaving this try block as it was, and throw a completely separate `RESOURCE_NOT_FOUND` exception straight after we've tried to retrieve the collection exercise (if it's null). That would keep it explicit and a bit cleaner.

Reply:
Yes I have reintroduced the original try block and moved the resource not found error to get called after the collection exercise call is made.